presto-main/src/main/java/com/facebook/presto/sql/rewrite/ShowStatsRewrite.java

```diff
@@ -287,19 +270,25 @@ private TableHandle getTableHandle(ShowStats node, QualifiedName table)
                     .collect(toImmutableList());
         }
 
-        private List<Expression> buildStatisticsRows(TableStatistics tableStatistics, Map<ColumnHandle, String> columnNames, Map<ColumnHandle, Type> columnTypes)
+        private List<Expression> buildStatisticsRows(TableHandle tableHandle, TableStatistics tableStatistics, Map<String, ColumnHandle> columns)
         {
             ImmutableList.Builder<Expression> rowsBuilder = ImmutableList.builder();
-
-            // Stats for columns
-            for (Map.Entry<ColumnHandle, ColumnStatistics> columnStats : tableStatistics.getColumnStatistics().entrySet()) {
-                ColumnHandle columnHandle = columnStats.getKey();
-                rowsBuilder.add(createColumnStatsRow(columnNames.get(columnHandle), columnTypes.get(columnHandle), columnStats.getValue()));
-            }
-
+            columns.forEach((columnName, columnHandle) -> {
+                ColumnMetadata columnMetadata = metadata.getColumnMetadata(session, tableHandle, columnHandle);
+                if (columnMetadata.isHidden()) {
+                    return;
+                }
+                Type columnType = columnMetadata.getType();
+                ColumnStatistics columnStatistics = tableStatistics.getColumnStatistics().get(columnHandle);
+                if (columnStatistics != null) {
+                    rowsBuilder.add(createColumnStatsRow(columnName, columnType, columnStatistics));
+                }
+                else {
+                    rowsBuilder.add(createEmptyColumnStatsRow(columnName));
```

Comment:
Could you add a test for a case when some columns (e.g. partition columns, maybe) don't have stats?

Reply:
That's one of the reasons why am i changing this. Before we were always assume that the `TableStatistics` object contains the statistics for all the columns. And it was true before the refactor. After refactor it changes. So it is implicitly tested by the existing tests.